urlddedf61e88_dwendelen_CG2_1_tgz-pJ8-RendProjectJ8/src/Color/RGBColor.java

```diff
@@ -0,0 +1,198 @@
+    mult.b = d.b * c.b;
+    return(mult);
+  }
+
+  /** Multiply color with a scalar, results in a new RGBColor object
+   * @param s scalar
+   * @return new RGBColor = this color *s
+   */
+  public RGBColor scale(double s) {
+    RGBColor mult = new RGBColor();
+    mult.r = this.r * s;
+    mult.g = this.g * s;
+    mult.b = this.b * s;
+    return(mult);
+  }
+
+  /** Multiply this color with a scalar
+   * @param s scalar
+   */
+  public void scaleSelf(double s) {
+    this.r *= s;
+    this.g *= s;
+    this.b *= s;
+
+  }
+
+  /** Divide color by a scalar, and create new RGBColor as a result
+   * @param s scalar
+   * @return new RGBCOlor = this color / s
+   */
+  public RGBColor scaleInverse(double s) {
+    RGBColor div = new RGBColor();
+    div.r = this.r / s;
+    div.g = this.g / s;
+    div.b = this.b / s;
+    return(div);
+  }
+
+  /** Print out color values, followed by newline
+   */
+  public void printlnRGBColor() {
+    System.out.println("Color comps: "+this.r+" -- "+this.g+" -- "+this.b);
+  }
+
+  /** Make a copy of the current color
+   * @return new object, copy of current one
+   */
+  public RGBColor makeCopy() {
+    RGBColor tmp = new RGBColor(this.r, this.g, this.b);
+    return(tmp);
+  }
+
+  /** Make a linear combination of given scalars and colors.
+   * Resulting color = a*color1 + b*color2
+   * @param a weight for color1
+   * @param color1 1st color
+   * @param b weight for color 2
+   * @param color2 2nd color
+   * @return resulting color, new object
+   */
+  public static RGBColor linearCombination(double a,RGBColor color1,double b,RGBColor color2) {
+    RGBColor tmp = new RGBColor();
+    tmp.r = a*color1.r + b*color2.r;
+    tmp.g = a*color1.g + b*color2.g;
+    tmp.b = a*color1.b + b*color2.b;
+    return tmp;
+  }
+
+  /** Make a linear combination of given scalars and colors.
+   * Resulting color = a*color1 + b*color2 + c*color3
+   * @param a weight for color1
+   * @param color1 1st color
+   * @param b weight for color 2
+   * @param color2 2nd color
+   * @param c weight for color 2
+   * @param color3 3nd color
+   * @return resulting color, new object
+   */
+  public static RGBColor linearCombination(double a,RGBColor color1,double b,RGBColor color2,double c,RGBColor color3) {
+    RGBColor tmp = new RGBColor();
+    tmp.r = a*color1.r + b*color2.r + c*color3.r;
+    tmp.g = a*color1.g + b*color2.g + c*color3.g;
+    tmp.b = a*color1.b + b*color2.b + c*color3.b;
+    return tmp;
+  }
+  /** Make a linear combination of given scalars and colors.
+   * Resulting color = color1 + b*color2
+   * @param color1 1st color
+   * @param b weight for color 2
+   * @param color2 2nd color
+   * @return resulting color, new object
+   */
+  public static RGBColor linearCombination(RGBColor color1,double b,RGBColor color2) {
+    RGBColor tmp = new RGBColor();
+    tmp.r = color1.r + b*color2.r;
+    tmp.g = color1.g + b*color2.g;
+    tmp.b = color1.b + b*color2.b;
+    return tmp;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 160:164, 174:178.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.